### ISSUE TITLE
Add a flag to disable download warnings when Safe Browsing is OFF

### ISSUE THREAD

**Issue by RonnyTNL** [brave/brave-browser#28917]
## Description 
Downloading .exe files now prompts for every file "This type of file can harm your computer" dialog (keep/discard).
Safe browsing is set to "No protection"

## Steps to Reproduce
Download any .exe file from which ever site

   1. download, download will start and dialog "This type of file can harm your computer" will appear.
   2. 
   3. 

## Actual result:
As safe browsing is set to disabled AND this did not happen in the past something has either changed or is broken.

## Expected result:
File should have been downloaded without being interrupted by the Keep/Discard dialog.

## Reproduces how often: 
100% over multiple machines

## Brave version (brave://version info)
1.48.171 Chromium: 110.0.5481.177 (Official Build) (64-bit)

## Version/Channel Information:
N/A

- Can you reproduce this issue with the current release? 
Yes
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 
No does not produce in Chrome.

**Comment by RonnyTNL** [comment]
Interestingly if I ENABLE Safe Browsing the issue disappears.

**Comment by RonnyTNL** [comment]
Has this changed in latest update then? this didn't happen on previous builds.

**Comment by RonnyTNL** [comment]
@fmarier Update on repro: 

I've verified with Chrome Stable (110.0.5481.178 (Official Build) (32-bit)) and Canary (113.0.5635.0 (Official Build) canary (64-bit)).

If you set Safe browsing to No Protection the default is to prompt for Keep/Discard.
However if you also switch "Ask where to save each file" this behavior is NOT seen and the download finishes uninterrupted.

That's the reason I noticed the change in behavior as I have both settings non default.

**Comment by fmarier** [comment]
@RonnyTNL Does this mean that the behavior the same between Chrome and Brave then?

**Comment by fmarier** [comment]
Thanks for the extra details @RonnyTNL . The reason why this started to happen is that we've recently made it so that the same checks are applied to downloads regardless of the "Ask where to save each file" setting: https://github.com/brave/brave-browser/issues/28079

So while in Chrome, you can suppress this behavior by:
- disabling Safe Browsing
- enabling "Ask where to save each file"

in Brave, that's no longer possible because we want to err on the side of protecting from more threats.

It's an unfortunate side-effect that disabling Safe Browsing means every `.exe` downloaded on Windows will throw up this warning, but since Safe Browsing is enabled by default and the warning can be bypassed by clicking the "Keep" button, I think it's a reasonable trade-off.

**Comment by fmarier** [comment]
Not sure `closed/wontfix` is the best label for this, because it's a legit issue, but there's no `closed/unfortunate-side-effect`.

**Comment by fmarier** [comment]
@RonnyTNL If you don't mind me asking, would you be willing to share why you disabled Safe Browsing? There are a lot of misconceptions about it and we've tried to improve some of the privacy properties in Brave, but I'd be curious to hear what made you disable it if you're willing to share.

**Comment by RonnyTNL** [comment]
Hi @fmarier 

First thing, there is a configuration option to disable something, then I expect that it does what it claims.

Setting it to "No protection" means "From now on I'm responsible" in my idea, so it kind of defeats the purpose to keep prompting from my point of view. And I get that from a use case of an average user downloads an occasional binary.
But if you chose to ignore that from a design perspective I guess you could remove the "No protection" setting completely.

I research security related stuff, might want to run in to an exploit, as we design anti-exploit protection, access stuff that Safe browsing prohibits, and download dozens of binaries a day, so in that case having to click on "Keep" get's rather annoying within a matter of minutes and frustrates my work, so I would really welcome some form of solution here (doesn't need to be UI, flag is fine) as long as I can disable this nag.

So it has nothing to do with not trusting the Safe browsing, I have a completely different use-case.

On a side-note, it seems there is still some part of the code that doesn't touch this "protection"
I have one way/link download that get saved without a prompt being an .exe file, but I guess it would be better not to post that in public.

**Comment by fmarier** [comment]
I see. Thanks for the details on your use-case. I agree that it's pretty annoying for those who have legitimate reasons for disabling this protection.

Now that I think more about it, I suspect that the warning might go away once https://github.com/brave/brave-browser/issues/17616 is fixed. That's the component that determines whether or not a file is (very loosely) "executable" and many of the checks and warnings are tied to that.

> On a side-note, it seems there is still some part of the code that doesn't touch this "protection" I have one way/link download that get saved without a prompt being an .exe file, but I guess it would be better not to post that in public.

You can email that to security@brave.com if you'd like.

**Comment by RonnyTNL** [comment]
Update, on latest release this now has a nasty effect, the download bar on the bottom is gone, which leaves you without any visible clue as to that the download was not finished. (There is no keep/discard notification drawing attention) as the download now seems to use the download bubble, which leaves you with a fully downloaded file only in the tmp format (Unconfirmed 123456.crdownload).

**Comment by fmarier** [comment]
Thanks for the update @RonnyTNL . I also noticed others reporting this with `.dmg` files on Mac: https://www.reddit.com/r/firefox/comments/12qrcl3/brave_browser_is_now_blocking_firefox_downloads/.

I think it's showing up enough in the wild that we need to address it.

**Comment by fmarier** [comment]
Some screenshots to illustrate the problem.

This is Brave with Safe Browsing turned off treating all `.exe` downloads as dangerous (including the Brave browser itself):
![Screenshot from 2023-04-21 15-10-07](https://user-images.githubusercontent.com/167821/233742289-33359dc8-16d7-418d-bc4a-2d241ef65630.png)

While this behavior can be worked-around in Chrome via the *Ask where to save each file* setting, if you disable Safe Browsing in Chrome, all `.exe` are treated as dangerous just like in Brave:
![Screenshot from 2023-04-21 15-09-09](https://user-images.githubusercontent.com/167821/233742419-e9d59705-c94d-435e-8c16-917b249cd5e0.png)
![Screenshot from 2023-04-21 15-09-41](https://user-images.githubusercontent.com/167821/233742420-ea66c416-667e-47b1-a9fe-52b352223bb6.png)

As @RonnyTNL said, now that the download notification bar is gone in Brave, this is even less obvious and it just looks like the download is not working.

In both Chrome and Brave, the downloads can be manually allowed by going into `chrome://downloads` and clicking *Keep*:
![Screenshot from 2023-04-21 15-15-31](https://user-images.githubusercontent.com/167821/233742664-323d3fe6-6a78-4f60-8501-53119b86f8bb.png)
![Screenshot from 2023-04-21 15-15-53](https://user-images.githubusercontent.com/167821/233742666-1635c2df-246f-426e-af63-d8f463b678d0.png)

**Comment by chuckmckinnon** [comment]
Chiming in -- I'm setting up a new computer and, my word, on _every single download_ I'm being told it's "dangerous" and have to open the download icon and click Keep. 

Notepad++ is dangerous now? VLC? Really? Because those are the prompts I'm getting.

I'm old. I started programming in Apple ][ days and have been using Windows since before antivirus was a thing. I value Brave because it protects my privacy; I disabled Safe Browsing because I don't want or need anyone (even Brave) nannying me and yes, I am _perfectly comfortable_ assuming any "risks" attendant on that decision. I'm not so stupid as to download files from illegitimate sources and if I do, that's _my_ problem.

Seriously, please make this stop. Like Upgrad3 last week, I'm considering finding alternatives to Brave for a daily driver because every. blessed. time. I have to tell it to Keep a file I just told it to download. Please stop trying to anticipate my security needs and let _me_ decide. Thank you.

**Comment by alexios-y** [comment]
Have been bothered with the same issue. I assume we can be responsible for our own behaviour when we deliberately choose to disable safe browsing feature. Why do you still block file download? Seriously considering changing browser too

**Comment by fmarier** [comment]
FYI, I do agree there is a need for a flag and this is on my TODO list. I've not been able to get to it as quickly as I expected for various reasons, but I'm not ignoring your use-case.

**Comment by andrebu** [comment]
So I had to find this thread because Brave team decided to protect me from my own explicit behavior, without giving me an option to turn off this horrible "protection"?  

Seriously, what are people thinking?  Hasn't every sane person had enough of this kind of presumptuous, controlling behavior? 

Thank you for the browser, Brave team, but no thanks for handicapping it and deciding for me what is and is not dangerous.

**Comment by fmarier** [comment]
> Brave team decided to protect me from my own explicit behavior

This warning is inherited from Chromium, it was not an explicit decision on our part. You also get the same behavior in Chrome by turning off Safe Browsing and also disabling "enabling "Ask where to save each file".

**Comment by Douglas-Klos** [comment]
This has been annoying the hell out of me for a couple months now, finally tried looking for a fix and ended up here.  I read inherited from Chromium.  Regardless, any idea on when this might be fixed?  Going to search for an older version of Brave and block updates so I don't have to deal with this.  Otherwise I love the browser.

**Comment by manfromarce** [comment]
While this behaviour is inherited from Chromium, the decision to show the prompt even when "Ask where to save each file" is enabled came from Brave. Also, Edge doesn't do this even if SmartScreen is disabled so it's definitely possible to override it.

**Comment by fmarier** [comment]
A new flag named `brave-override-download-danger-level` will be available in Brave Nightly shortly, and then make it way to the 1.54 release.

**Comment by pixelstuff** [comment]
I agree with some of the others. The expected behavior doesn't make sense. Perhaps there needs to be three options in the settings.

- Maximum Protection: Checks URLs and blocks all unsafe downloads with no override.
- Minimum Protection: Checks URLs and asks to Keep/Discard any unsafe downloads.
- No Protection: Doesn't double check anything or ask about unsafe downloads.

I don't know if the idea needs to be kicked back up to the original Chromium project or just implemented separately in Brave.

**Comment by Fasto5** [comment]
This STILL hasn't been addressed. 
Why can't we have 'safe browsing' and CHOOSE to disable the STUPID 'exe' download warning option. 
I'm not a software engineer - but I install / delete programs I deem safe for testing all nearly every day I use a computer.
It is nanna-state-infuriating the way it is

**Comment by fmarier** [comment]
@Fasto5 As of Brave 1.54 (currently in Beta):

1. If you have Safe Browsing ON, then the warning will be suppressed for safe downloads (i.e. most of them).
2. If you have Safe Browsing OFF, then you can enable the `brave-override-download-danger-level` flag in `brave://flags` to suppress the warning in all cases.

Note that #1 will be improved further when https://github.com/brave/brave-browser/issues/26183 makes it to release (in Brave 1.55).

**Comment by stephendonner** [comment]
## Verification `PASSED` using

### Steps:

1. installed `1.57.16`
2. launched Brave
3. opened `brave://settings/security`
4. toggled `Safe Browsing` to `No protection (not recommended)`
5. confirmed the warning dialog
6. loaded `https://zoom.us/download`
7. clicked on `Download`
8. clicked on `Save`
9. confirmed that the download was blocked as dangerous
10. set `brave://flags/#brave-override-download-danger-level` to `Enabled`
11. clicked `Relaunch`
12. reloaded `https://zoom.us/download`
13. clicked on `Download`
14. clicked `Save`

### Confirmed that the download was saved successfully, without any warnings/block/confirmation

example | example | example | example | example | example | example | example
---------|---------|----------|----------|---------|----------|---------|----------
<img width="1312" alt="Screen Shot 2023-07-18 at 12 56 37 PM" src="https://github.com/brave/brave-browser/assets/387249/82310e44-5e48-437e-b0a4-2ef62bb80843"> | <img width="1312" alt="Screen Shot 2023-07-18 at 12 56 41 PM" src="https://github.com/brave/brave-browser/assets/387249/bc574e5c-48e9-4d7e-8367-ab71d4936d02"> | <img width="1312" alt="Screen Shot 2023-07-18 at 12 58 25 PM" src="https://github.com/brave/brave-browser/assets/387249/caa861b0-400b-4099-a041-4760972cff12"> | <img width="1312" alt="Screen Shot 2023-07-18 at 12 58 34 PM" src="https://github.com/brave/brave-browser/assets/387249/3a8ba7f5-3357-4490-84b9-97ff399797ba"> | <img width="1312" alt="Screen Shot 2023-07-18 at 1 00 36 PM" src="https://github.com/brave/brave-browser/assets/387249/97d2215b-0c5f-4c5c-a707-19b100273acf"> | <img width="1312" alt="Screen Shot 2023-07-18 at 1 04 54 PM" src="https://github.com/brave/brave-browser/assets/387249/49aaa170-3fcc-442a-9d87-7d3975d07291"> | <img width="1312" alt="Screen Shot 2023-07-18 at 1 06 06 PM" src="https://github.com/brave/brave-browser/assets/387249/8c9b494c-b67b-4dc5-a8b9-f706acf919c5"> | <img width="1312" alt="Screen Shot 2023-07-18 at 1 06 10 PM" src="https://github.com/brave/brave-browser/assets/387249/3e7b0dd2-c210-4a53-ac2e-5e0512cf68d6">

**Comment by alexios-y** [comment]
Currently on 1.56.11, the latest version. However, still can't find this flag to disable downloading warnings. Can I ask when it's going to be released?

**Comment by fmarier** [comment]
@alexios-y We ended up having to bump the version numbers by 3 a few weeks ago due to a technical glitch. I've updated my comment above. The flag will be in 1.57.

**Comment by fmarier** [comment]
Just noting that Edge has apparently the same behavior: https://textslashplain.com/2023/08/23/smartscreen-application-reputation-in-pictures/

**Comment by levicki** [comment]
How long is this flag going to be maintained?

I gave a pretty good use case and rationalization as to why those warnings are pointless in the feature request where I asked for an option to whitelist domains and filetypes like it is possible in Microsoft Edge. Is it possible that nobody has read that?

**Comment by levicki** [comment]
> If you don't mind me asking, would you be willing to share why you disabled Safe Browsing?

I am a bit late to this discussion, but I think you might want to hear another angle:

Said prompt (Keep / Discard) is:

**Annoying** &mdash; It is totally indiscriminate (for example it warns you that NVIDIA driver setup can harm your computer), and IMO it just leads to user fatigue, which in turn leads to user being trained to click Keep / OK / Yes buttons without thinking.

**Pointless (at least on Windows platform)** &mdash; Windows has built-in antivirus since 2013. Furthermore, all downloaded files are saved with [Mark Of The Web](https://textslashplain.com/2016/04/04/downloads-and-the-mark-of-the-web/), so the user will get a warning from Windows Explorer when they try to open the file / run the program.

**Useless** &mdash; Unlike the browser prompt, the operating system prompt at least provides some information (application publisher and whether it has a valid digital signature) which can help the user to decide whether to proceed or not. Browser keep / discard prompt is just a scare prompt which doesn't provide any information that can be acted upon.

In Microsoft Edge, they have added a policy setting where one can whitelist file extensions and domains so that warning is not displayed when downloading files with whitelisted extensions from whitelisted domains.

**Comment by fmarier** [comment]
> In Microsoft Edge, they have added a policy setting where one can whitelist file extensions and domains so that warning is not displayed when downloading files with whitelisted extensions from whitelisted domains.

That's a good idea. If you'd like to propose that in a separate issue, we can track it there.

**Comment by Fasto5** [comment]
> > In Microsoft Edge, they have added a policy setting where one can whitelist file extensions and domains so that warning is not displayed when downloading files with whitelisted extensions from whitelisted domains.
> 
> That's a good idea. If you'd like to propose that in a separate issue, we can track it there.

Why should we have to spend our time 'whitelisting' websites?

-why can't we just have an option to disable the reminder re EXE file type?

(afterall there is no warning for zip files, which can contain exe files!)

**Comment by levicki** [comment]
@fmarier 
> > In Microsoft Edge, they have added a policy setting where one can whitelist file extensions and domains so that warning is not displayed when downloading files with whitelisted extensions from whitelisted domains.
> 
> That's a good idea. If you'd like to propose that in a separate issue, we can track it there.

That [issue](https://github.com/brave/brave-browser/issues/30285) is already open for 5 months &mdash; you commented on it and didn't notice it is a feature request?

**Comment by levicki** [comment]
@Fasto5 
> -why can't we just have an option to disable the reminder re EXE file type?

You already have that &mdash; brave://flags/#brave-override-download-danger-level

> Why should we have to spend our time 'whitelisting' websites?

That's not the kind of whitelist you would be adding to non-stop, 99% of people don't go and download every possible exe file they can find on the Internet. It is meant for enterprise admins who want to whitelist safe domains and filetypes (not just exe).

The way Microsoft Edge implemented whitelisting makes it possible to also whitelist everything on every domain which would be the same as disabling it.

**Comment by fmarier** [comment]
> That [issue](https://github.com/brave/brave-browser/issues/30285) is already open for 5 months — you commented on it and didn't notice it is a feature request for that functionality?

I've updated the title of that issue to make it clearer.

**Comment by guest271314** [comment]
> @RonnyTNL If you don't mind me asking, would you be willing to share why you disabled Safe Browsing? There are a lot of misconceptions about it and we've tried to improve some of the privacy properties in Brave, but I'd be curious to hear what made you disable it if you're willing to share.

Because they turned the "feature" off. The developer and user shouldn't have to further explain why they don't want a feature enabled. That's what the option, when clearly written, provides the developer and user the choice to do.

> So while in Chrome, you can suppress this behavior by:
> 
> - disabling Safe Browsing
> - enabling "Ask where to save each file"
> - 
> in Brave, that's no longer possible because we want ...

Wow. 

I found this issue after turning off Google Safe Browsing on Chrome-For-Testing Version 121.0.6103.0 (Official Build) (64-bit). ["Insecure download blocked": I didn't ask Chromium/Chrome to block downloads and erroneously list a .deb file as "insecure""](https://groups.google.com/a/chromium.org/g/chromium-dev/c/n_m4I5prB14/).

Google Safe Browsing folks claim the download "warning" has nothing to do with Google Safe Browsing. 

Interestingly in this post the term 

> whitelisted 

is being used where such arbitrary color coding terminology has no place in programming. [Issue 842296: Use inclusive terminology throughout Chromium](https://bugs.chromium.org/p/chromium/issues/detail?id=842296).

> > By the way, why is Google Safe Browsing still using the term "whitelist" here safebrowsing.safe_browsing_whitelist_domains:?
>
> Preferences are persisted to storage. Renaming them is non-trivial and potentially error-prone and requires maintaining migration code for long periods of time.

Folks who implement Safe Browsing and the like "features" sure are opinioned about what _they want_. Yet don't get around to cleaning up using color coding laguage in legacy code where "white" is "good" and "black" is "bad".

I think source code authors need to provide _options_ for developers and users to disable whatever alleged "security" and "privacy" measures _they_ think are good for _themselves_; the maintainer can enable the feature on their own machine, and the developer and user can disable the feature on their own machine. That's reasonable.

I sure wasn't expected what I read here in Brave world. Sign of the times I guess.

**Comment by guest271314** [comment]
Looks like https://github.com/chromium/chromium/blob/585a487dde15771a986d9e5a027768a75a34789f/chrome/browser/about_flags.cc#L8821

```
--disable-features=InsecureDownloadWarnings
```

disables the "Insecure download blocked" prompt on Chromium.

**Comment by guest271314** [comment]
Just when I thought I had gotten rid of the download warnings somebody came up with yet a _different_ warning message.

Chromium Version 121.0.6109.0 (Developer Build) (64-bit) downloaded this morning with 

```
--disable-features=InsecureDownloadWarnings
```

flag set and Google Safe Browsing turned OFF is now giving a different warning on HTTP: (http://archive.ubuntu.com/ubuntu/pool/main/b/build-essential/)

> Unverified download blocked.

This is insane.

**Comment by guest271314** [comment]
We have to enable

> Ask where to save each file before downloading

for 

```
--disable-features=InsecureDownloadWarnings
```

to work.

Keep in mind none of these combinations of flags and settings to disable the download warnings is spelled out in Google Chrome or Chromium documentation.

If you think all roads lead back to Google Safe Browsing there are yet other "features" that are in play trying to prevent you from downloading the file you decide to download on the Web page (HTTP:; FILE:; whatever) that you decide to visit.

**Comment by root-phantasoft** [comment]
> @RonnyTNL If you don't mind me asking, would you be willing to share why you disabled Safe Browsing? There are a lot of misconceptions about it and we've tried to improve some of the privacy properties in Brave, but I'd be curious to hear what made you disable it if you're willing to share.

Because it is my computer, not yours. I, the user, make the decisions, and I don't want the software making decisions for me. 

sudo rm -rf / means delete all the bloody files on my computer. Don't ask me, don't tell me if I'm sure, don't make choices for me. 

All of this "features" really insult the intelligence of a lot of users, specially when they can't be completely disabled. I don't ever want to see a warning, I don't want a confirmation dialog, I don't need my own computer second guessing my actions.

**Comment by sorquan** [comment]
Chrome is getting worse and worse, the freedom to customize the old chromium is disappearing somewhere. Why can’t we just make at least a flag to disable this useless protection that no one needs?

**Comment by fmarier** [comment]
> Why can’t we just make at least a flag to disable this useless protection that no one needs?

There is a flag: `brave://flags/#brave-override-download-danger-level`

**Comment by MNLierman** [comment]
> **Chrome** is getting worse and worse, the freedom to customize the old **chromium** is disappearing somewhere. Why can’t we just make at least a flag to disable this useless protection that no one needs?

> There is a flag: `brave://flags/#brave-override-download-danger-level`

He said Chrome and Chromium, not Brave. I know this issue is in /brave/ but he specifically said Chromium. I think a lot of people are here in this discussion thread because of a Reddit post referencing this stupid Chromium behavior. Stupid Google, come on. Have A/B tests actually shown that crap UI decisions like this stop the spread of malware? By creating extra steps a user will have to click through once they turn off Safe Browsing? Yeah, I don't think so.
@fmarier @sorquan

**Comment by Ghrem** [comment]
> > **Chrome** is getting worse and worse, the freedom to customize the old **chromium** is disappearing somewhere. Why can’t we just make at least a flag to disable this useless protection that no one needs?
> 
> > There is a flag: `brave://flags/#brave-override-download-danger-level`
> 
> He said Chrome and Chromium, not Brave. I know this issue is in /brave/ but he specifically said Chromium. I think a lot of people are here in this discussion thread because of a Reddit post referencing this stupid Chromium behavior. Stupid Google, come on. Have A/B tests actually shown that crap UI decisions like this stop the spread of malware? By creating extra steps a user will have to click through once they turn off Safe Browsing? Yeah, I don't think so. @fmarier @sorquan

I've been arguing with _the incredibly useless and ignorant morons_ that are Google One support over this issue and they know absolutely nothing. Probably take an act of god to get them to truly allow disabling this ridiculous blocking on their end. That flag Brave added should be pushed up into Chromium.

**Comment by guest271314** [comment]
@Ghrem See above at `--disable-features=InsecureDownloadWarnings`

- https://github.com/chromium/chromium/blob/3ae33ca6c8d566139bd18e281d1fa0a97ae79c23/chrome/browser/download/insecure_download_blocking.cc#L468
- https://github.com/chromium/chromium/blob/3ae33ca6c8d566139bd18e281d1fa0a97ae79c23/chrome/common/chrome_features.cc#L1294
- https://github.com/chromium/chromium/blob/3ae33ca6c8d566139bd18e281d1fa0a97ae79c23/chrome/browser/flag_descriptions.h#L1180

**Comment by levicki** [comment]
@fmarier The flag `Override download danger level` no longer works in `1.60.114 Chromium: 119.0.6045.124 (Official Build) (64-bit)` &mdash; browser is asking whether to keep or discard a `.dll` file.

**Comment by fmarier** [comment]
@levicki It seems to work for me. Here's what I did:

1. Start Brave with a fresh profile.
2. Set the _Safe Browsing_ setting to **No protection** in `chrome://settings/security`.
![Screenshot from 2024-01-22 15-13-45](https://github.com/brave/brave-browser/assets/167821/d1f07af3-87e5-487f-9ba2-cdf521ed385f)
3. Set the _Override download danger level_ flag to **Enabled** in `brave://flags/#brave-override-download-danger-level`:
![Screenshot from 2024-01-22 14-57-58](https://github.com/brave/brave-browser/assets/167821/69cca047-2d78-4102-8306-0b95ae99705b)
4. Restart the browser.
5. Go to https://zoom.us/download#client_4meeting and download the Zoom client.

## Result

Following the above steps, the download is not blocked:
![Screenshot from 2024-01-22 15-11-22](https://github.com/brave/brave-browser/assets/167821/9aff82ea-69ec-42c4-b5fc-30ee7bf592f2)

On the other hand, when the _Override download danger level_ flag is set to **Default** (or **Disabled**), then I get the warning:
![Screenshot from 2024-01-22 14-20-22](https://github.com/brave/brave-browser/assets/167821/cfaa6788-9de9-4954-a1be-8709423e688d)
![Screenshot from 2024-01-22 14-20-30](https://github.com/brave/brave-browser/assets/167821/59ed95cd-a104-4ad8-9f93-c8f9cd7ede86)

## Versions tested

I tested:
- Stable channel 1.61.120 Chromium: 120.0.6099.234 on Linux and Windows 10
- Beta channel 1.63.126 Chromium: 121.0.6167.57 on Linux
- Nightly channel 1.64.12 Chromium: 121.0.6167.75 on Linux

all with the same results, though of course the file was an `.exe` for Windows.

**Comment by fmarier** [comment]
@levicki If you're still seeing something different from me, can you try again with 1.61 and then tell me:
- the URL of a website / download file where you can reproduce the problem
- what you have the Safe Browsing setting and the `brave://flags/#brave-override-download-danger-level` flag set to

**Comment by guest271314** [comment]
@fmarier Nice work in Brave world if you got this far. From Chromium Version 122.0.6251.0 (Developer Build) (64-bit) with Google Safe Browsing turned off. I'll have to try some disabling feature flags, again. This is insane. 

![Screenshot_2024-01-22_17-39-17](https://github.com/brave/brave-browser/assets/4174848/f78e5d26-081d-4c28-b2d7-930469254804)

**Comment by fmarier** [comment]
@guest271314 I'm not aware of a way to opt out of this behavior on Chrome/Chromium.

**Comment by guest271314** [comment]
Interesting.

Somebody baked a "warning" about a `.deb` file in to WICG File System Access API

![Screenshot_2024-01-22_21-22-10](https://github.com/brave/brave-browser/assets/4174848/8f3432fd-c0ef-4850-b930-f3d47604e597)

but not a Chromium `.zip` download from https://download-chromium.appspot.com/

![Screenshot_2024-01-22_21-26-19](https://github.com/brave/brave-browser/assets/4174848/d9073288-e293-46ec-bc96-d98681092ce5)

**Comment by levicki** [comment]
@fmarier I updated to latest release just now.

Here is the URL: https://github.com/p0358/Fuck_off_EA_App/releases

Try downloading `version.dll` file.

What happens for me:

![image](https://github.com/brave/brave-browser/assets/16415478/dd395738-22a8-471d-aee5-593b8b8cb3c0)

![image](https://github.com/brave/brave-browser/assets/16415478/5b110f28-828d-4958-b547-75c248d49c5d)

My settings:

Safe browsing

![image](https://github.com/brave/brave-browser/assets/16415478/ef1885ba-d8ca-4d19-9e7a-d67d50f961ec)

Flags:

![image](https://github.com/brave/brave-browser/assets/16415478/4a8ff5e7-6e94-4404-9434-08c212c1e011)

**Comment by fmarier** [comment]
@levicki I found the problem: https://github.com/brave/brave-browser/issues/35561. It seems like there are only 6 file types that have the "DANGEROUS" danger level and `.exe` (or any of the ones I had previously tested) is not one of those.

@guest271314 The warnings are platform-specific and are defined in https://source.chromium.org/chromium/chromium/src/+/main:components/safe_browsing/content/resources/download_file_types.asciipb, a file that's downloaded as part of the "File Type Policies" component you can see in `chrome://components`.

**Comment by guest271314** [comment]
@fmarier Well, something ain't working as written out here https://source.chromium.org/chromium/chromium/src/+/main:components/safe_browsing/content/resources/download_file_types.asciipb;l=3439-3448.

The user gesture is the click on the link (HTML element)

```
file_types {
  extension: "deb"
  uma_value: 141
  ping_setting: FULL_PING
  platform_settings {
    platform: PLATFORM_TYPE_LINUX
    danger_level: ALLOW_ON_USER_GESTURE
    auto_open_hint: DISALLOW_AUTO_OPEN
  }
}
```

Chromium doesn't have anything in `chrome://components`, no components installed. Just like there are no terms (TOS) in `chrome://terms`.

> "File Type Policies"

That got me to thinking about policies, i.e., `chrome://policy`, where there is at least one relevant entry https://chromeenterprise.google/policies/?policy=ExemptDomainFileTypePairsFromFileTypeDownloadWarnings

> [ { "file_extension": "jnlp", "domains": ["example.com"] }, { "file_extension": "exe", "domains": ["example.com"] }, { "file_extension": "swf", "domains": ["*"] } ]
> 
> Note that while the preceding example shows the suppression of file type extension-based download warnings for "swf" files for all domains, applying suppression of such warnings for all domains for any dangerous file type extension is not recommended due to security concerns. It is shown in the example merely to demonstrate the ability to do so.

**Comment by fmarier** [comment]
> Well, something ain't working as written out here https://source.chromium.org/chromium/chromium/src/+/main:components/safe_browsing/content/resources/download_file_types.asciipb;l=3439-3448.
> 
> The user gesture is the click on the link (HTML element)

In Chromium, a danger level of `ALLOW_ON_USER_GESTURE` triggers the warning in the download manager when Safe Browsing is OFF. That's why you're seeing this (on the Linux platform).

> Chromium doesn't have anything in `chrome://components`, no components installed.

[This fallback code](https://source.chromium.org/chromium/chromium/src/+/main:components/safe_browsing/content/common/file_type_policies.cc;l=75-83;drc=13eb2077904eeb3bb460642b1b6c0c9e31942346) probably applies then.

Keep in mind though, I'm not a Chromium expert. I've only ever tested this in Brave.

**Comment by levicki** [comment]
@fmarier Thanks for looking into it, glad you could reproduce it.

I think that having `ini` and `cfg` listed as dangerous is plain ridiculous.

Moreover, `manifest` could probably be narrowed down to `dll.manifest` and `exe.manifest` so that other files with `manifest` extension are unaffected.

Is there a chance you could provide at least a rough ETA for the fix to hit the release channel?

**Comment by levicki** [comment]
> I have a [PR in review](https://github.com/brave/brave-core/pull/21760) which should land in time for 1.64 (currently scheduled to hit stable on March 19th).

Good to know.

Out of curiosity, why not change `cfg`, `dll`, and `ini` to be `ALLOW_ON_USER_GESTURE` level as well? Are they seriousl that more dangerous than `exe`?

**Comment by guest271314** [comment]
>  It's one thing to add a layer of security

There really is no such thing as "security" for any signal communications. As of last century certain entities were analyzing 20 TB per second via ThinThread. PRISM, Apple "disclosing" the had an agreement with the U.S. Government to not disclose to users agreements between that concern and the U.S. Government re user data, et al.

**Comment by fmarier** [comment]
> Out of curiosity, why not change `cfg`, `dll`, and `ini` to be `ALLOW_ON_USER_GESTURE` level as well?

That list is maintained by the Google Safe Browsing team and we use it without modifications in Brave (proxied of course). It's not currently something we have had the need to fork in Brave yet.

**Comment by digitaldreamer7** [comment]
Thanks for providing one more annoyance that we can't get rid of.  I don't need you or google to hold my hand.  My av can handle anything that's downloaded.  This browser is becoming more and more frustrating for people who don't need YOU to decide what WE want to do on our own personal computers.

**Comment by luckman212** [comment]
Here's why _THIS FEATURE FUC*ING SUCKS_

Yesterday I was at a customer's office dealing with an emergency Network Down situation. The firewall's flash memory had become corrupt and I had to access it using a console cable in a very difficult to reach location. It was a tense moment but I got it repaired. Our policy is to save copies of the running config for these types of situations. So I clicked the "download config" button, saw the download complete (or so I thought), climbed down from the ladder, packed up my bag and left.

This morning, went to upload the config to our server for safekeeping and saw the file never made it to my disk. It was blocked & canceled because it was "insecure". THANKS GOOGLE.

**Comment by guest271314** [comment]
FWIW This might be the place to lodge your feedback on the record for download warnings [Q4 2023 Summary from Chrome Security](https://groups.google.com/a/chromium.org/g/chromium-dev/c/9NQ8PSXkXVI).

**Comment by levicki** [comment]
@digitaldreamer7 @luckman212 Guys you are barking up the wrong tree here. Being abusive towards Brave devs doesn't really help &mdash; you are antagonizing the only people who can actually help us by making a browser which doesn't fully follow Google's (and Chromium's) bullshit policies like this one.

In case you didn't read the full thread and just came to vent your anger here is a short summary:

1. When you turn off Safe Browsing all Chromium based browsers block downloads that are considered dangerous. What is dangerous is decided by Google Safe Browsing Team, not Brave devs.
2. Brave users submitted a feature request for the ability to disable download blocking and that was implemented via [brave://flags/#brave-override-download-danger-level](brave://flags/#brave-override-download-danger-level) flag.
3. However, because of an oversight (which I reported here), that flag doesn't fully disable download blocking for all file tyes &mdash; the fix for that is currently scheduled for release in version 1.64 on March 19.

So, if you are using Brave, you just need to be a bit patient. Thanking Brave devs for honoring a feature request and putting in the extra work to override the default Chromium behavior in order to allow us to disable download blocking wouldn't hurt either.

**Comment by luckman212** [comment]
Sorry - my frustration wasn't towards Brave or the devs who are working towards finding a good solution to this. Yes I was venting a bit—towards Google—because it was being discussed. I meant no disrespect.

**Comment by 825i** [comment]
Chromium browsers are such dogsh*te these days. I as the owner of my own computer and as an IT professional of 15+ years, am not allowed to download a freaking PDF FILE without being harassed every time! Because for some INSANE reason even on an HTTPS website, it's deemed "insecure". 

Besides shouldn't the word in this case be UNSECURE? I don't quite think my .pdf file suffers from crippling depression or some other kind of imposter syndrome...

![image](https://github.com/user-attachments/assets/098cd2e9-21de-40f5-b52b-2c4ed2afd624)

But yes, apparently being allowed to completely and totally disable garbage like this, is just not possible on the Chrome ecosystem. Wow. One step closer to a free and open internet aye folks?

Make no mistake, I have nothing against the Brave devs and I otherwise love this browser. Yet at the ass end of 2024, not having a way to disable this kind of thing is just plain Orwellian at this stage.

**Comment by digitaldreamer7** [comment]
> I as the owner of my own computer and as an IT professional of 15+ years, am not allowed to download a freaking PDF FILE without being harassed every time! Because for some INSANE reason even on an HTTPS website, it's deemed "insecure".
> 
> Besides shouldn't the word in this case be UNSECURE? I don't quite think my .pdf file suffers from crippling depression or some other kind of imposter syndrome...
> 
> ![image](https://private-user-images.githubusercontent.com/58630825/366492215-098cd2e9-21de-40f5-b52b-2c4ed2afd624.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwODc5MjUsIm5iZiI6MTcyNjA4NzYyNSwicGF0aCI6Ii81ODYzMDgyNS8zNjY0OTIyMTUtMDk4Y2QyZTktMjFkZS00MGY1LWI1MmItMmM0ZWQyYWZkNjI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTExVDIwNDcwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ0ZjBkOWEzNTYyODU1YWMxM2YyNDFjZjBmMDNmZjk4NmUxYTBmODY0N2E1MjAwYjMxYjc2OGFmYWZkNGFhZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IA5W0WDuErQyp_Hlj6g1bhAmuw0gcW-lVN9qmUIrsGw)
> 
> But yes, apparently being allowed to completely and totally disable garbage like this, is just not possible on the Chrome ecosystem. Wow. One step closer to a free and open internet aye folks?
> 
> Make no mistake, I have nothing against the Brave devs and I otherwise love this browser. Yet at the ass end of 2024, not having a way to disable this kind of thing is just plain Orwellian at this stage.

Seems like they have a grammar and a development issue

**Comment by pixelstuff** [comment]
> I as the owner of my own computer and as an IT professional of 15+ years, am not allowed to download a freaking PDF FILE without being harassed every time! Because for some INSANE reason even on an HTTPS website, it's deemed "insecure".
> 
> Besides shouldn't the word in this case be UNSECURE? I don't quite think my .pdf file suffers from crippling depression or some other kind of imposter syndrome...

I don't think UNSECURE is an official word. INSECURE just means not sure or certain, or not adequately protected. The word itself isn't intrinsically tied to someone's mental state anymore than the word blue is.

**Comment by guest271314** [comment]
FWIW Here's an unpacked extension that uses Native Messaging to launch Cloudflare's `workerd` locally, providing the capability to write and read files directly to the local file system https://github.com/guest271314/native-messaging-workerd. To bypass arbitrary browser restictions, limitations, prompts, warnings, altogether. E.g., in DevTools `console`

```
 await (await fetch(`http://localhost:8081/${location.href.split("/").pop()}`, {
  method: "put",
  body: await (await fetch(location.href)).text()
})).text()
'File written to ~/Downloads/ws.js'
```

**Comment by 825i** [comment]
> > I as the owner of my own computer and as an IT professional of 15+ years, am not allowed to download a freaking PDF FILE without being harassed every time! Because for some INSANE reason even on an HTTPS website, it's deemed "insecure". Besides shouldn't the word in this case be UNSECURE? I don't quite think my .pdf file suffers from crippling depression or some other kind of imposter syndrome...
> 
> I don't think UNSECURE is an official word. INSECURE just means not sure, not certain, or not adequately protected. The word itself isn't intrinsically tied to someone's mental state anymore than the word blue is.

Staight from the Oxford dictionary my friend

![image](https://github.com/user-attachments/assets/c8ee9280-c7a3-4572-a014-5bd282bd0be6)

Here's insecure

![image](https://github.com/user-attachments/assets/4b308527-d309-472d-a898-43b96cc693f5)

FYI, this bit was more tongue-and-cheek but after getting bombarded with this popup whenever I try to download handfuls of .pdf files, I felt compelled to have a stab at how stupid the wording is on top of everything else.

Another amazing find today. I'm able to download multiple .exe files from an HTTP unsecure connection that is served through an app on my phone, giving me access to my phone's storage. Without being prompted that these .exe files could be dangerous.

So...

* .exe and HTTP = OKAY.
* .pdf and HTTPS = NOT OKAY.

_Wow_ yeah... this is definitely working as intended.

**Comment by levicki** [comment]
> I as the owner of my own computer and as an IT professional of 15+ years, am not allowed to download a freaking PDF FILE without being harassed every time! Because for some INSANE reason even on an HTTPS website, it's deemed "insecure".

Given that you are such a professional, are you absolutely sure it's HTTPS? What does it say for this file for example?

https://us.download.nvidia.com/Windows/561.09/561.09-win11-win10-release-notes.pdf

I have Safe Browsing disabled and for me this PDF is not being blocked by Brave.

> Besides shouldn't the word in this case be UNSECURE? I don't quite think my .pdf file suffers from crippling depression or some other kind of imposter syndrome...

[Unsecure](https://www.merriam-webster.com/dictionary/unsecure) isn't a word despite what that Google AI generated dictionary excerpt based on AI generated Google SEO gaming fluff being indexed is telling you.

[Insecure](https://www.etymonline.com/word/insecure) on the other hand quite literally means not safe.

> Staight from the Oxford dictionary my friend

Straight from the lying LLM mouth my friend:

![image](https://github.com/user-attachments/assets/5ad13384-704b-4bc6-8035-56a86f56d209)

> EDGE at this point is less restrictive and more user friendly out of the box than brave.... EDGE...

Then stop using Brave and use Edge instead of harrasing people here.

**Comment by pixelstuff** [comment]
> > > I as the owner of my own computer and as an IT professional of 15+ years, am not allowed to download a freaking PDF FILE without being harassed every time! Because for some INSANE reason even on an HTTPS website, it's deemed "insecure". Besides shouldn't the word in this case be UNSECURE? I don't quite think my .pdf file suffers from crippling depression or some other kind of imposter syndrome...
> > 
> > 
> > I don't think UNSECURE is an official word. INSECURE just means not sure, not certain, or not adequately protected. The word itself isn't intrinsically tied to someone's mental state anymore than the word blue is.
> 
> Staight from the Oxford dictionary my friend
> (screenshot)
> 
> Here's insecure
> (screenshot)
> 

That is pretty weird. Oxford seems to be the only dictionary to include "unsecure" and when I search in Oxford myself it says it is an obsolete word last used in the early 1700s. Why would they reference a "computer system" in an obsolete word? Merriam-Webster, Cambrige, and Dictionary.com don't mention unsecure at all.

With insecure most of them include additional definitions. Here are some from the Merriam-Webster Dictionary.
![Insecure](https://github.com/user-attachments/assets/97152c06-e977-4c12-a3b8-95c5984534ef)

**Comment by levicki** [comment]
> Oxford seems to be the only dictionary to include "unsecure" and when I search in Oxford myself it says it is an obsolete word last used in the early 1700s. Why would they reference a "computer system" in an obsolete word?

That's an AI generated blurb and AI hallucinates BS all the time.

There's quite a number of articles debating "insecure vs unsecure" in Google search results &mdash; they are probably skewing the answer by giving the "unsecure" word some statistical validity in ranking.

TL;DR &mdash; see the screenshot I posted from ollama3.1 model above, it first claims unsecure is a word, then when I confront the answer it admits it's not a word according to dictionary. Likely the training data was poisoned.

**Comment by Targren** [comment]
Maybe we should get back to focusing less vocabulary wanking and more trying to determine how we can make it so that the browser isn't stepping on me when I try to download a .deb package on my linux box, after I explicitly told I don't want it to protect me.

**Comment by guest271314** [comment]
Just use an extension, policies, and/or Native Messaging and be done with the matter. 

Then you don't have to ask maintainers to do or not do anything. You implement the strategy yourself.

**Comment by guest271314** [comment]
While Chromium is arbitrarily popping up download warnings `AI` is now shipped in Chromium Version 131.0.6746.0 (Developer Build) (64-bit), defined on the global object. 

Who asked for that? Chromium decided to download `AI` for you...

How about block "AI" marketing on every surface known to humankind?

Too funny.

**Comment by MNLierman** [comment]
Guys we have a couple of options to making this a reality, and also
changing other things that we want. It's obvious that Brave won't do it
themselves.

I know a bit of coding, and there are people I'm sure reading these
comments that also know coding. Who is interested in joining a group of
people who know at least some coding and can build an extension or fork? We
have 2 options, #2 is likely what we will have to do, but maybe not. We can
put together a Discord group and  review options together and decide.

1. Build an extension that overrides this default behavior. A flag or
option created/modified by an extension should be fine. Manifest v3 will
break this, and I don't know if v2 will allow, haven't researched it yet.

2. If an extension won't work, maybe a patcher, or our own Brave builds.
GitHub automatic build actions ***SHOULD*** be able to ***automately** *build
a Brave, and we can replace the updater to point to our repo. Isn't Brave
completely open source buildable by anyone? The heart and inspiration of
the Chromium project insists that all forks are automatically buildable and
automatic GitHub actions should not break on code changes.

As long as Brave follows this rule then we should be able to create
automatic builds that run every time Brave makes changes, or on a weekly
basis, with instant builds on high priority security patches. It can build
automatically in GitHub Ubuntu containers for any architecture and it's
**free**. If anyone is interested and knows anything about coding, then
let's get together and do this!

**Comment by guest271314** [comment]
@MNLierman I already created such a Web extension and shared it here https://github.com/brave/brave-browser/issues/28917#issuecomment-2345110479. Specifically for downloading and creating directories without any prompts or user-activation permissions requests. Somebody marked it as off-topic. MV3 is not a challenge to bend to the will of the programmer.

**Comment by whateverusernameisnottaken** [comment]
Please add me to the growing number of people who would like to see this change. After having worked in IT on and off for decades, I find the status quo both annoying and cumbersome.

**Comment by Nenormalniy** [comment]
Guys, on your main page it says “The browser that puts you first”. But in reality, the browser treats me like a nanny treats a baby or a dog being walked on a leash. It's up to you to decide, but with this approach you'll end up distrusting the browser the same way you stopped trusting Chrome.

**Comment by Ghrem** [comment]
This fucking problem still exists and the devs are pieces of fucking shit. Go fuck yourselves.

**Comment by whateverusernameisnottaken** [comment]
Thankfully, you don't speak for any of the rest of us. Your name calling and extreme rudeness have no place here. These people work hard all the time on software we get for free. And you think you have the right to say things like that? Smarten up or go away.

**Comment by xamindar** [comment]
This stupid "keep" prompt has caused me to miss a backup zip I thought I downloaded before I wiped my device and rebuilt it......only to find this stupid nag prompt made for a child had quietly ignored my file download request.

Just when I thought Brave was different and giving control to the user, it SNATCHES it away with this boneheaded option that caused me to lose a lot of time and work because it DOESN'T DO WHAT IT WAS ASKED TO DO!

I'm sorry, but this option is completely moronic. When the user clicks to tell it to download something, and it silently ignores that request with a little tiny "keep" word hidden up in the corner on a freaking 4K display, instead of doing what the user expects.......data loss is the result.

Very disappointing to see this has not ben fixed and instead closed as "wont fix/we do not trust any users".

**Comment by logiczny** [comment]
> This stupid "keep" prompt has caused me to miss a backup zip I thought I downloaded before I wiped my device and rebuilt it......only to find this stupid nag prompt made for a child had quietly ignored my file download request.
> 
> Just when I thought Brave was different and giving control to the user, it SNATCHES it away with this boneheaded option that caused me to lose a lot of time and work because it DOESN'T DO WHAT IT WAS ASKED TO DO!
> 
> I'm sorry, but this option is completely moronic. When the user clicks to tell it to download something, and it silently ignores that request with a little tiny "keep" word hidden up in the corner on a freaking 4K display, instead of doing what the user expects.......data loss is the result.
> 
> Very disappointing to see this has not ben fixed and instead closed as "wont fix/we do not trust any users".

same story here, what a fucked world we live in. Software decides for us what is better for me and double checks my actions.
I was sure that I downloaded a backup of my network switch, then, day later, I have none, after I've made big changes to it. Thanks developers for that fucked up options that does nothing and is turned ON by default.

**Comment by whateverusernameisnottaken** [comment]
What's the most offensive to me is that downloads from my OWN NETWORK  are being blocked for my safety. This makes zero sense, as if you can't trust devices on your own LAN, you're already screwed. I too have missed downloads because I didn't notice the prompt. Please, take the nanny mentality out of this and let us control things. Thanks.

**Comment by guest271314** [comment]
> What's the most offensive to me is that downloads from my OWN NETWORK are being blocked for my safety. This makes zero sense, as if you can't trust devices on your own LAN, you're already screwed. I too have missed downloads because I didn't notice the prompt. Please, take the nanny mentality out of this and let us control things. Thanks.

Write and use Web extensions to download content.

**Comment by whateverusernameisnottaken** [comment]
> > What's the most offensive to me is that downloads from my OWN NETWORK are being blocked for my safety. This makes zero sense, as if you can't trust devices on your own LAN, you're already screwed. I too have missed downloads because I didn't notice the prompt. Please, take the nanny mentality out of this and let us control things. Thanks.
> 
> Write and use Web extensions to download content.

I don't know what that involves, but I doubt I'm up to it. Anyways, I want a simpler experience, not more complexity.

**Comment by guest271314** [comment]
> > > What's the most offensive to me is that downloads from my OWN NETWORK are being blocked for my safety. This makes zero sense, as if you can't trust devices on your own LAN, you're already screwed. I too have missed downloads because I didn't notice the prompt. Please, take the nanny mentality out of this and let us control things. Thanks.
> > 
> > 
> > Write and use Web extensions to download content.
> 
> I don't know what that involves, but I doubt I'm up to it. Anyways, I want a simpler experience, not more complexity.

Clearly asking the maintainers for a "simpler experience" ain't happening. 

Once you write your own Web extensions you can essentially do whatever you want in these browsers. 

Here's one way I wrote to download content directly to the file system. One using `node`, and a branch that uses QuickJS https://github.com/guest271314/native-messaging-file-writer.

**Comment by Ghrem** [comment]
Ridiculous work around for a ridiculous problem. Whoever closed this problem needs fucking shot in the face with a salt load.

**Comment by Ghrem** [comment]
> Thanks for the extra details [@RonnyTNL](https://github.com/RonnyTNL) . The reason why this started to happen is that we've recently made it so that the same checks are applied to downloads regardless of the "Ask where to save each file" setting: [#28079](https://github.com/brave/brave-browser/issues/28079)
> 
> So while in Chrome, you can suppress this behavior by:
> 
> * disabling Safe Browsing
> * enabling "Ask where to save each file"
> 
> in Brave, that's no longer possible because we want to err on the side of protecting from more threats.
> 
> It's an unfortunate side-effect that disabling Safe Browsing means every `.exe` downloaded on Windows will throw up this warning, but since Safe Browsing is enabled by default and the warning can be bypassed by clicking the "Keep" button, I think it's a reasonable trade-off.

You're a douche bag. Fix the original problem, dipshit

**Comment by whateverusernameisnottaken** [comment]
> Ridiculous work around for a ridiculous problem. Whoever closed this problem needs fucking shot in the face with a salt load.

Look, we're not happy about this. But I suggest you get some perspective. These people work night and day to make a good browser. Nothing really compares to it. Trashing them is neither ethical nor helpful. Suggest you cool off.

**Comment by Ghrem** [comment]
> > Ridiculous work around for a ridiculous problem. Whoever closed this problem needs fucking shot in the face with a salt load.
> 
> Look, we're not happy about this. But I suggest you get some perspective. These people work night and day to make a good browser. Nothing really compares to it. Trashing them is neither ethical nor helpful. Suggest you cool off.

I work day and night with engineers and I, nor our CEO, would accept this kind of behavior. It's stupid, abnormal, and being unable to disable it is tenfold stupidity.

**Comment by xamindar** [comment]
> > Ridiculous work around for a ridiculous problem. Whoever closed this problem needs fucking shot in the face with a salt load.
> 
> Look, we're not happy about this. But I suggest you get some perspective. These people work night and day to make a good browser. Nothing really compares to it. Trashing them is neither ethical nor helpful. Suggest you cool off.

It's not a good browser with this feature, unfortunately. The fact they closed the issue thinking it was a good decision makes me suspect they are making boneheaded decisions in other aspects of the product. I will not stick with a browser that takes away control from the user.

**Comment by whateverusernameisnottaken** [comment]
Fine, then use another one, and please stop spewing anger here.

**Comment by guest271314** [comment]
@whateverusernameisnottaken 

> Nothing really compares to it.

Huh?

```
wget --show-progress --progress=bar --output-document chrome.zip \ 
https://download-chromium.appspot.com/dl/Linux_x64?type=snapshots \ 
&& unzip chrome.zip && rm chrome.zip
```

**Comment by whateverusernameisnottaken** [comment]
Then if you feel Chrome is a viable alternative, then use it, and stop complaining. Geez.

**Comment by guest271314** [comment]
> Then if you feel Chrome is a viable alternative, then use it, and stop complaining. Geez.

That's to fetch the tip-of-tree Chromium. What Brave, Opera, Edge are made _from_. 

I don't complain. I roll my own and `FixWontFix`.

The 

> Nothing really compares to it.

claim is simply false.
